File: java/src/main/java/io/eventdriven/slimdownaggregates/slimmed/BookEvent.java
package io.eventdriven.slimdownaggregates.slimmed;

import io.eventdriven.slimdownaggregates.slimmed.entities.*;

public sealed interface BookEvent {
  record ChapterAdded(
    BookId BookId,
    Chapter Chapter
  ) implements BookEvent {
  }

  record Approved(
    BookId BookId,
    CommitteeApproval CommitteeApproval
  ) implements BookEvent {
  }

  record MovedToEditing(
    BookId BookId
  ) implements BookEvent {
  }

  record MovedToPrinting
    (
      BookId BookId
    ) implements BookEvent {
  }

  record Published(
    BookId BookId,
    ISBN ISBN,
    Title Title,
    Author Author
  ) implements BookEvent {
  }

  record BookMovedToOutOfPrintEvent
    (
      BookId BookId
    ) implements BookEvent {
  }

  record FormatAdded(
    BookId BookId,
    Format Translation
  ) implements BookEvent {
  }

  record FormatRemoved(
    BookId BookId,
    Format Translation
  ) implements BookEvent {
  }

  record TranslationAdded(
    BookId BookId,
    Translation Translation
  ) implements BookEvent {
  }
}

File: csharp/SlimDownYourAggregates.Tests/Original/BookEvent.cs
using SlimDownYourAggregates.Tests.Original.Entities;

namespace SlimDownYourAggregates.Tests.Original;

public record WritingStarted(
    BookId BookId,
    Genre Genre,
    Title Title,
    Author Author,
    ISBN ISBN
);

public record ChapterAdded(
    BookId BookId,
    Chapter Chapter
);

public record FormatAdded(
    BookId BookId,
    Format Format
);

public record FormatRemoved(
    BookId BookId,
    Format Format
);

public record TranslationAdded(
    BookId BookId,
    Translation Translation
);

public record MovedToEditing(
    BookId BookId
);

public record Approved(
    BookId BookId,
    CommitteeApproval CommitteeApproval
);

public record MovedToPrinting
(
    BookId BookId
);

public record Published(
    BookId BookId,
    ISBN ISBN,
    Title Title,
    Author Author
);

public record MovedToOutOfPrint
(
    BookId BookId
);
